### PR TITLE
[WIP] Fix Knex adapter to not automatically drop/init the DB

### PR DESCRIPTION
**Pull request by molomby**
Updates to knex DB adapter to..

* Separate code used to create the schema from the `postConnect` function
* Remove logic that (incorrectly) auto-detects when the DB needs to be setup (and nukes existing data)

This is short term fix for #2095.

### CODE REVIEW

**Comment by MadeByMike** [comment]
This means to seed the database initially you need to run with `dropDatabase`. This is going to flood us with support requests.

**Comment by molomby** [comment]
@MadeByMike Yeah, you're not wrong.. I suppose we should make some corresponding updates to docs/guides?

I'll mark this PR as WIP in the meantime.

**Comment by MadeByMike** [comment]
As a first step can we push something that says this NEVER runs unless `process.env.NODE_ENV === 'development'`

**Comment by gautamsi** [comment]
can we not generate migration script for knex based on changes to lists?

 perhaps we have keystone util which can generate knex migration scripts based on list config snapshots, it will also give opportunity to anyone to modify the migration scripts to add initial seeding data.

File: packages/adapter-knex/lib/adapter-knex.js [hunk 1]
```diff
@@ -68,14 +68,17 @@ class KnexAdapter extends BaseKeystoneAdapter {
...
       listAdapter._postConnect();
     });
-    const isSetup = await this.schema().hasTable(Object.keys(this.listAdapters)[0]);
-    if (this.config.dropDatabase || !isSetup) {
-      console.log('Knex adapter: Dropping database');
-      await this.dropDatabase();
-    } else {
+
+    // Drop and recreate the list table, if it's been explicitly requests
+    if (!this.config.dropDatabase) {
```
Comment (MadeByMike) [inline packages/adapter-knex/lib/adapter-knex.js]: Leave is setup and check if env is 'development'. Then fix the long term prob.

File: .changeset/loud-trains-relax.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@keystonejs/adapter-knex': patch
```
Comment (MadeByMike) [inline .changeset/loud-trains-relax.md]: This is a breaking change

**Pull request closed without merging**